### PR TITLE
discretization: low order cell basis can be evaluated on all type of cells

### PR DESCRIPTION
**Pull request by andrea-iob**
Polynomials of degree 0 and 1 are supported on all type of cells, for pixels and voxels also degree 3 is supported.

Note: coefficients are evaluated as the volume average of the Taylor series expansion coefficients. In order to support higher order degrees on all type of cells, we will need to implement numerical integration of the expansions terms over the volume of the cell. This can be done using Gauss quadrature rules and evaluating the terms on the integration points using the function that evaluate the basis function on a specified point.

### CODE REVIEW

**Review by lucask81 — APPROVED** [review]

**Review by lucask81 — APPROVED** [review]

**Comment by andrea-iob** [comment]
For some reasons the review done by @lucask81 is not enough to allow me to merge the pull request. @marcocisternino Can you have a look at these changes?

**Comment by andrea-iob** [comment]
(We require a review from a user that has write access and @lucask81 doesn't.)

File: src/discretization/reconstruction.cpp
Comment (marcocisternino) [inline src/discretization/reconstruction.cpp]: There's a discrepancy between doc and code. Max supported degree in code is 2, while doc says 3.

Comment (marcocisternino) [reply src/discretization/reconstruction.cpp]: I pushed a fix

**Review by marcocisternino — APPROVED** [review]

**Pull request merged**
